modules/core/src/main/scala/lucuma/odb/api/repo/ConstraintSetRepo.scala

```diff
@@ -53,13 +47,15 @@ object ConstraintSetRepo {
         oid:            Observation.Id,
         includeDeleted: Boolean
       ): F[Option[ConstraintSetModel]] =
-        tablesRef.get
-          .flatMap { tables =>
-            tables.constraintSetObservation.selectLeft(oid).traverse { csid =>
-              tryFindConstraintSet(tables, csid).liftTo[F]
-            }
-          }
-          .map(x => deletionFilter(includeDeleted)(x))
+        tablesRef.get.map{ t => 
+          val ocs = for {
+            o <- t.observations.get(oid)
+            csid <- o.constraintSetId
+            cs <- t.constraintSets.get(csid)
+          } yield cs
+
+          ocs.filter(cs => includeDeleted || cs.isPresent) 
```

Comment:
i wonder if it would work to put the filter in the `for` expression. 
```
  cs <- t.constraintSets.get(csid)
  if includeDeleted || cs.isPresent
} yield cs
```
No big deal but would save a line I suppose.

Reply:
Good idea.